_posts/2024-02-17-social-integration.md

```diff
@@ -0,0 +1,170 @@
+What we came up with, is a system where a plugin can only be loaded into OpenTTD, if we, the OpenTTD Development Team, approved the plugin.
+
+We find this less than ideal, as we like that BaNaNaS allows anyone to come up with ideas, try them out, see what people think.
+But, we also have to keep the security of players in mind.
+And there is just too much potential for harm when we would open up the plugin system to anyone.
+As such, we were left with no other choice than to restrict who can make those plugins; or of course, not do it at all (which is even less ideal).
+
+In result, every plugin is released and signed by OpenTTD, and before loaded into the game, it is validated the signature is correct.
+If you are a developer and want to design your own plugin, please come talk to us.
+We are not against new and different plugins; but we have a responsibility to the players to ensure it doesn't compromise their security.
+
+## Capabilities
+
+With the licensing and security finally addressed, and a solid idea how to build the plugin system, we started work.
+But as you can imagine, it was a lot of work, with a lot of moving parts.
+Like ... seriously, it took many hours figuring all this out, and getting it right.
+
+An additional challenge was that we release for three OSes: Windows, Linux and MacOS.
+And they all deal with plugins in a slightly different way; enough to be challenging.
+A lot of testing, failing, trying again went into getting this right.
+
+Which also meant we had to make choices: get something to work now, and build on it later, or try to do everything at once, potentially never finishing it at all.
+We went for the first approach, but it also means the current capabilities of the plugin system are rather limited.
+So don't think too much of it yet.
+
+The main thing it currently does is announce you are playing the game, whether you are in the Main Menu or in-game, and what kind of map-size you are playing.
+Other things like being able to join each others games etc is all not implemented yet, but hopefully someone will pick that up for the next version.
+But, as always, no promises there.
+
+## How does it work?
+
+When you first start OpenTTD 14, a new folder "social_integration" will be created in your OpenTTD documents folder.
+In here plugins can be installed, for example the [Steam plugin](https://www.openttd.org/downloads/steam-social-releases/latest).
+You have to extract its content in this "social_integration" folder, and start the game.
+
+When the game starts, it is validated that the plugins are signed off by OpenTTD, and if they are, they are loaded.
+The first step for the plugin is to check whether the social platform is running.
+If so, it starts the integration with that platform.
+A plugin can only integrate with a platform if it is running when OpenTTD starts; so make sure Steam, Discord, or GOG Galaxy is already running before you start OpenTTD.
```

Comment:
Should this be rephrased to say, "A plugin can only integrate with a platform that is running when OpenTTD starts; so make sure Steam, Discord, or GOG Galaxy is already running before you start OpenTTD." This change clarifies what "it" is (the platform not the plugin).

Reply:
Tnx!